src/core/container.jl

```diff
@@ -81,19 +72,20 @@ Data structure for particle filters
 - normalise!(pc::ParticleContainer)
 - consume(pc::ParticleContainer): return incremental likelihood
 """
-mutable struct ParticleContainer{T<:Particle, F, Tvals <: Array{T}, TlogW <: Array{Float64}}
-    model :: F
-    num_particles :: Int
-    vals  :: Tvals
-    logWs :: TlogW       # Log weights (Trace) or incremental likelihoods (ParticleContainer)
-    logE  :: Float64     # Log model evidence
-    # conditional :: Union{Nothing,Conditional} # storing parameters, helpful for implementing rejuvenation steps
-    conditional :: Nothing # storing parameters, helpful for implementing rejuvenation steps
```

Comment:
Thanks for cleaning this up - this was meant to implement the rejuvenation step in the SMC2 algorithm, but the `VarInfo` mechanism turns out to be a more generic mechanism for evaluation conditional distributions (e.g. those in Gibbs).